hikari/impl/bot.py

```diff
@@ -861,20 +862,22 @@ async def start(
                 )
             )
 
-            for started_shard in started_shards:
-                self._shards[started_shard.id] = started_shard
-
         await self.dispatch(lifetime_events.StartedEvent(app=self))
 
         _LOGGER.info("application started successfully in approx %.2f seconds", time.monotonic() - start_time)
 
+    def _check_if_alive(self) -> None:
+        if self._closing_event.is_set():
+            raise errors.ComponentNotRunningError("bot is not running so it cannot be interacted with")
```

Comment:
how would someone looking in from out be able to programmatically tell that a shard or client is closing or closed?

Reply:
Added an `is_alive` property to the bot. Thx for the spot